Libraries/HyperV.psm1

```diff
@@ -186,21 +199,22 @@ Function CreateAllHyperVGroupDeployments($setupType, $xmlConfig, $Distro, $Debug
                                     LogMsg "Test Platform is $TestPlatform and Test Area is $TestArea, need to enable nested virtualization"
                                     $status = EnableHyperVNestedVirtualization -HyperVGroupName $HyperVGroupName -HyperVHost $HyperVHost
                                 }
-                                $StartVMStatus = StartHyperVGroupVMs -HyperVGroupName $HyperVGroupName -HyperVHost $HyperVHost
-
-                                if ($StartVMStatus)
-                                {
-                                    $retValue = "True"
-                                    $isHyperVGroupDeployed = "True"
-                                    $HyperVGroupCount = $HyperVGroupCount + 1
-                                    $DeployedHyperVGroup += $HyperVGroupName
-                                }
-                                else 
-                                {
-                                    LogErr "Unable to start one or more VM's"
-                                    $retryDeployment = $retryDeployment + 1
-                                    $retValue = "False"
-                                    $isHyperVGroupDeployed = "False"
+                                foreach ($HyperVHost in $HyperVHostArray){
+                                    $StartVMStatus = StartHyperVGroupVMs -HyperVGroupName $HyperVGroupName -HyperVHost $HyperVHost
+                                    if ($StartVMStatus)
+                                    {
+                                        $retValue = "True"
+                                        $isHyperVGroupDeployed = "True"
+                                        $HyperVGroupCount = $HyperVGroupCount + 1
+                                        $DeployedHyperVGroup += $HyperVGroupName
+                                    }
+                                    else 
+                                    {
+                                        LogErr "Unable to start one or more VM's"
+                                        $retryDeployment = $retryDeployment + 1

```

Comment:
is there a retry deployment? if so, mentioned it in the message, or what does this do?